train_colors_weaksup.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import print_function
```

Comment:
this is okay for now but try to reuse code as much as possible -- lots of duplicate here and in `train_colors.py`